### PR TITLE
Config manager

### PR DESCRIPTION
**Pull request by yarovyii**

### CODE REVIEW

**Comment by eric3873** [comment]
Please test if parameters are good (not empty depending of scan type ...):
https://github.com/42ity/fty-discovery/blob/00ec2218ffcec3e0613b2c77d41203abf234686b/src/fty_discovery_server.cc#L392

Please add management of local scan:
https://github.com/42ity/fty-discovery/blob/00ec2218ffcec3e0613b2c77d41203abf234686b/src/fty_discovery_server.cc#L239

Please initialize the ip address list (m_listIpAddress) with input parameters (ip list, range scan list, local scan, netmask list...)
https://github.com/42ity/fty-discovery/blob/00ec2218ffcec3e0613b2c77d41203abf234686b/src/fty_discovery_server.cc#L392
https://github.com/42ity/fty-discovery-ng/blob/743ed1b46b20fc510ef0a84fa3b58367a05bf76d/server/src/jobs/auto-discovery.h#L60

File: common/discovery-config-manager.cpp [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+        return fty::unexpected("discovery config was not loaded => nothing to save");
+    }
+    m_config.value().save(path);
+
+    return {};
+}
+
+fty::Expected<ConfigDiscovery> ConfigDiscoveryManager::load(const std::string& path)
+{
+    auto tmp = ConfigDiscovery();
```
Comment (etnKlendathu) [inline common/discovery-config-manager.cpp:23]: Process description:
Create object;
Create second object;
Assign (possible as copy) to first one...
Of course, will be optimized, but better ti write:
```ConfigDiscovery tmp;``` looks better and no questions.

File: common/discovery-config-manager.cpp [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+void ConfigDiscoveryManager::set(const ConfigDiscovery& config)
+{
+    m_config = config;
+}
+
+fty::Expected<void> ConfigDiscoveryManager::commandCreate(const commands::config::Config& in)
+{
+    auto tmp = ConfigDiscovery();
```
Comment (etnKlendathu) [inline common/discovery-config-manager.cpp:47]: Again, who imagine it?

File: common/discovery-config-manager.cpp [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    return {};
+}
+
+bool ConfigDiscoveryManager::validateIp(const std::string& ip)
+{
+    // return std::regex_match(ip,
+    // std::regex("(?=^.{4,253}$)(^((?!-)[a-zA-Z0-9-]{0,62}[a-zA-Z0-9]\\.)+[a-zA-Z]{2,63}$)"));
+    return std::regex_match(
+        ip, std::regex("(^((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\\.){3}(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$)"));
```
Comment (etnKlendathu) [inline common/discovery-config-manager.cpp:190]: make std::regex as static...

**Review by eric3873 — APPROVED** [review]

**Pull request merged**